receiver/mongodbreceiver/scraper_test.go

```diff
@@ -297,15 +238,16 @@ func TestScraperScrape(t *testing.T) {
 			scraper := newMongodbScraper(componenttest.NewNopReceiverCreateSettings(), createDefaultConfig().(*Config))
 			scraper.client = tc.setupMockClient(t)
 			actualMetrics, err := scraper.scrape(context.Background())
-
 			if tc.expectedErr == nil {
 				require.NoError(t, err)
 			} else {
 				if strings.Contains(err.Error(), ";") {
 					// metrics with attributes use a map and errors can be returned in random order so sorting is required.
-					actualErrs := strings.Split(err.Error(), ";")
+					// The first error message would not have a leading whitespace and hence split on "; "
+					actualErrs := strings.Split(err.Error(), "; ")
 					sort.Strings(actualErrs)
-					expectedErrs := strings.Split(tc.expectedErr.Error(), ";")
+					// The first error message would not have a leading whitespace and hence split on "; "
+					expectedErrs := strings.Split(tc.expectedErr.Error(), "; ")
```

Comment:
These changes appear unrelated.

Reply:
It is unrelated but necessary. 

The first error string in both actual and expected error has a missing leading whitespace due to choice of delimiter. This change has updated delimiter to ensure all errors follow same format.

should this be a different PR?